### PR TITLE
annotate-snippets: update to 0.10

### PR DESCRIPTION
**Pull request by klensy**
Ports `annotate-snippets` to 0.10, temporary dupes versions; other crates left that depends on 0.9 is `ui_test` and `rustfmt`.

### CODE REVIEW

**Comment by davidtwco** [comment]
`anstyle` seems fine to introduce as a transitive dependency. Is there a reason you couldn't just bump `rustfmt`'s use here too?

**Comment by klensy** [comment]
> `anstyle` seems fine to introduce as a transitive dependency. Is there a reason you couldn't just bump `rustfmt`'s use here too?

I'll send pr to rustfmt repo directly, so less moving parts when syncing with rustfmt.

File: compiler/rustc_errors/src/annotate_snippet_emitter_writer.rs [hunk 1]
```diff
@@ -190,11 +189,6 @@ impl AnnotateSnippetEmitter {
                     annotation_type: annotation_type_for_level(*level),
                 }),
                 footer: vec![],
-                opt: FormatOptions {
-                    color: true,
```
Comment (klensy) [inline compiler/rustc_errors/src/annotate_snippet_emitter_writer.rs:194]: Was color: true, so styled() now?

**Comment by klensy** [comment]
Pushed fix plus updated to 0.10.1 (https://github.com/rust-lang/annotate-snippets-rs/blob/0.10.1/CHANGELOG.md#0101---2024-01-04)

**Comment by davidtwco** [comment]
Ping me once you've got CI passing :)

**Review by davidtwco — APPROVED** [review]

**Pull request merged**